### DISCUSSION TITLE
Does undefined do nothing if i use the some, every, none operators?

### DISCUSSION THREAD

**Discussion by ADrejta** [prisma/prisma#8963] [Q&A] [answered]
Hi,

I think when using undefined with the some, every, none operators in a one to many relationship it does not operate in the same way that is described in [this](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined#the-effect-of-null-and-undefined-on-conditionals) guide

Example:
Let's say we have users and each user can have many posts and I want to retrieve users that have a certain title of their post.
```js
await prisma.users.findMany({
         where : {
                posts: {
                   some: {
                      title: {
                         contains: searchText
                      }
                   }
              }
         }
     })
```

In cases where a user does not have any posts and i send the `searchText` variable as undefined that user is not returned.
Is that the intended behaviour?

**Comment by ryands17** [comment, marked as answer]
@ADrejta 👋 
Yes that is the intended behaviour. If you want all the users to be returned if `searchText` is `undefined`, then you need to do the following:

```ts
await prisma.users.findMany({
         where : {
                posts: {
                   some: searchText ? {
                       title: {
                         contains: searchText
                      }
                   } : {}
              }
         }
     })
```

**Reply by ADrejta** [reply]
Okay great!

I do think it is worth pointing this in the guide mentioned above as to not catch people like me off-guard 😅

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
